src/discord/client.rs

```diff
@@ -0,0 +1,167 @@
+                exit(-1);
+            }
+        }
+    }
+
+    async fn interaction_create(&self, ctx: Context, interaction: Interaction) {
+        let start = Instant::now();
+
+        let labels = match interaction {
+            Interaction::Ping(_)
+            | Interaction::MessageComponent(_)
+            | Interaction::Autocomplete(_)
+            | Interaction::ModalSubmit(_) => {
+                vec![(LABEL_NAME_INTERACTION, interation_name(&interaction))]
+            }
+            Interaction::ApplicationCommand(ref command) => {
+                info!(
+                    "➡️ Received command interaction: {} ({}) from {}",
+                    command.data.name, command.id, command.user.name
+                );
+                debug!("🔬Command is: {:#?}", command);
+
+                let discord_command = DiscordCommand::from_str(&command.data.name);
+                let labels = vec![
+                    (LABEL_NAME_INTERACTION, interation_name(&interaction)),
+                    (
+                        LABEL_NAME_COMMAND,
+                        discord_command
+                            .as_ref()
+                            .map(|name| name.to_string())
+                            .unwrap_or_else(|_| LABEL_VALUE_COMMAND_UNKNOWN.to_string()),
+                    ),
+                ];
+
+                let execution_result: Result<(), DiscordError> = match discord_command {
+                    Ok(DiscordCommand::Ping) => {
+                        PingCmd {}.execute(&ctx, &interaction, command).await
+                    }
+                    _ => Err(DiscordError::from(UnknownCommand(format!(
+                        "🤔 I don't understand: {}",
+                        command.data.name
+                    )))),
+                };
+
+                match execution_result {
+                    Ok(_) => {
+                        info!("✅ Successful execute slash command");
+                    }
+                    Err(err) => {
+                        warn!("❌ Failed to execute command: {}", err);
+
+                        let content =
+                            format!("😖I failed to execute the command! (error was: {})", err);
+                        if let Err(why) = Self::send_response(&ctx, command, content).await {
+                            warn!("❌ Cannot respond to slash command: {}", why);
+                        }
+                    }
+                }
+
+                labels.to_vec()
+            }
+        };
+        let delta = start.elapsed();
+
+        increment_counter!(METRIC_DISCORD_INTERACTIONS_TOTAL, &labels);
+        histogram!(METRIC_DISCORD_INTERACTIONS_DURATION, delta, &labels);
+    }
+}
+
+impl Handler {
+    async fn send_response(
+        ctx: &Context,
+        command: &ApplicationCommandInteraction,
+        content: String,
+    ) -> serenity::Result<()> {
+        command
+            .create_interaction_response(&ctx.http, |response| {
```

Comment:
Maybe add the ephemeral flag to the bot reply
This prevent from pollute channel with bot command
https://docs.rs/serenity/latest/serenity/builder/struct.CreateInteractionResponseData.html#method.ephemeral

Reply:
Well, I don't think that's a problem in this case, the ping command is supposed to be removed eventually. It's just to test the bot that it was implemented. But concerning the next "token faucet" request, well it can be used. For successful requests, I think it's better that it's visible to everyone in the channel, but for error ones, we could use an ephemeral message... To be discussed later.

Thanks for the proposal.